browser/app/js/browser/MainActions.js

```diff
@@ -16,83 +16,151 @@
 
 import React from "react"
 import { connect } from "react-redux"
-import { Dropdown, OverlayTrigger, Tooltip } from "react-bootstrap"
 import web from "../web"
+import classNames from "classnames"
 import * as actionsBuckets from "../buckets/actions"
 import * as uploadsActions from "../uploads/actions"
-import { getPrefixWritable } from "../objects/selectors"
+import { getPrefixWritable, getCheckedList } from "../objects/selectors"
+import MakeBucketModal from "../buckets/MakeBucketModal"
+import ReactTooltip from "react-tooltip"
 
-export const MainActions = ({
-  prefixWritable,
-  uploadFile,
-  showMakeBucketModal
-}) => {
-  const uploadTooltip = <Tooltip id="tt-upload-file">Upload file</Tooltip>
-  const makeBucketTooltip = (
-    <Tooltip id="tt-create-bucket">Create bucket</Tooltip>
-  )
-  const onFileUpload = e => {
+export class MainActions extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      addNewActive: false,
+      makeBucketActive: false
+    }
+  }
+
+  onFileUpload(func, e) {
     e.preventDefault()
     let files = e.target.files
     let filesToUploadCount = files.length
     for (let i = 0; i < filesToUploadCount; i++) {
-      uploadFile(files.item(i))
+      func(files.item(i))
```

Comment:
Do we need to pass `func` to this function.
Instead can we just do `this.props.uploadFile`